apps/crn-server/src/migrations/1646992836-replace-empty-string-with-null-in-research-output-link.ts

```diff
@@ -0,0 +1,27 @@
+/* istanbul ignore file */
+import { RestResearchOutput } from '@asap-hub/squidex';
+import { Migration } from '../handlers/webhooks/webhook-run-migrations';
+import { applyToAllItemsInCollection } from '../utils/migrations';
+
+export default class MapResearchOutputDeprecatedSubtype extends Migration {
+  up = async (): Promise<void> => {
+    await applyToAllItemsInCollection<RestResearchOutput>(
+      'research-outputs',
+      async (researchOutput, squidexClient) => {
+        if (researchOutput?.data?.link?.iv === '') {
+          await squidexClient.patch(researchOutput.id, {
+            link: {
+              // @ts-expect-error // The type system does not support null here.
```

Comment:
i guess you're right, although for the purpose of this migration you could alter the type on line 8 to allow null for the link and then wouldn't need to ts-expect-error here

but i see your point, the types are correct for the GET endpoints in squidex but it's different with the patch

Reply:
I think hard-coding the type is more problematic than the `// @ts-expect-error`, because the expect-error indicates that the types are off, while hardcoding the types make it seem like everything is ok, while in truth we are casting to some specific type.